.github/workflows/component-builder.yml

```diff
@@ -0,0 +1,59 @@
+name: "Component Builder"
+on:
+  pull_request_target:
+    paths:
+      - 'containers/fedora/**'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  guest-fedora-amd64:
+    runs-on: ubuntu-latest
+    env:
+      FEDORA_IMAGE: Fedora-Cloud-Base-Generic-41-1.4.x86_64.qcow2
+      FEDORA_VERSION: 41
+      CPU_ARCH: amd64
+      FULL_EMULATION: "true"
+      DEBIAN_FRONTEND: noninteractive
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+      - name: Install dependencies for VM build
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+            qemu-system-x86 \
+            libvirt-daemon-system \
+            virtinst cloud-image-utils \
+            libguestfs-tools
+      - name: Tweak hosted runner to enable 'virt-sysprep'
+        # https://bugs.launchpad.net/ubuntu/+source/linux/+bug/759725
+        run: sudo chmod 0644 /boot/vmlinuz*
+      - name: Fetch base Fedora image
+        working-directory: ./containers/fedora
+        run: wget -q "https://download.fedoraproject.org/pub/fedora/linux/releases/41/Cloud/x86_64/images/${{ env.FEDORA_IMAGE }}"
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          version: "0.6.11"
+      - name: Create VM
+        working-directory: ./containers/fedora
+        env:
+          PYTHONPATH: ${{ github.workspace }}
+          ORGANIZATION_ID: ${{ secrets.BITWARDEN_ORGANIZATION_ID }}
+          ACCESS_TOKEN: ${{ secrets.BITWARDEN_ACCESS_TOKEN }}
+        run: ./build.sh
+      - name: Logging to quay.io
```

Comment:
I think we should add a verification step before pushing the image, to verify tat the build was successful. We should prevent unhealthy images to be pushed to Quay.

Reply:
You will have to explain what that means exactly and suggest what you expect to do.
First, it pushed to the staging repo, not to the production one, allowing verification against the staging before it is manually pushed to production. In fact, the follow up possible steps express it in the PR description.

Reply:
I think another action in pr merged to push to prod.

While the pr is open you will have the image on stage to verify it, once verified the pr is merged then the second action push it to prod.

WDYT?

Reply:
Yes, that makes sense.
But that will go all the way to automate it from change to deployment.
I think it will be safer to start with this one, then add `arm64` support and only everything is good, to add the second job.

BTW, it will be interesting if we can pass the container images between jobs or cache them.